pgmpy/factors/Factor.py

```diff
@@ -191,39 +191,26 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(),variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        sum_variable = np.zeros(len(factor.cardinality))
+        sum_variable[marginalize_index] = -1
```

Comment:
And also why are you creating two variables `assign` and `sum_variable`? I think you should be able to work with only one of them.